src/TiciTacaToeyGameEngine.ts

```diff
@@ -360,76 +514,120 @@ class TiciTacaToeyGameEngine implements GameEngine {
               .filter((each) => game.spectators.includes(each))
               .map((each) => this.players[each]);
             const response: Response = {
-              // todo: extract as generic method
               type: message.type,
-              game,
-              players: connectedPlayers
-                .map((each) => ({ name: each.name, playerId: each.playerId }))
-                .reduce((acc, each) => {
-                  acc[each.playerId] = each;
-                  return acc;
-                }, {}),
-              spectators: connectedSpectators
-                .map((each) => ({ name: each.name, playerId: each.playerId }))
-                .reduce((acc, each) => {
-                  acc[each.playerId] = each;
-                  return acc;
-                }, {}),
+              game: {
+                ...game,
+                timers: getTimerBaseFromGame(game),
+              },
+              players: getPlayers(connectedPlayers),
+              spectators: getSpectators(connectedSpectators),
             };
-            connectedPlayers.forEach((player) => {
-              player.connection.send(JSON.stringify(response));
-            });
-            connectedSpectators.forEach((player) => {
-              player.connection.send(
-                JSON.stringify({
-                  ...response,
-                  type: MessageTypes.SPECTATE_GAME,
-                })
-              );
-            });
+            sendResponseToPlayers(
+              MessageTypes.SPECTATE_GAME,
+              response,
+              connectedPlayers,
+              connectedSpectators
+            );
           });
         break;
       case MessageTypes.START_GAME:
       case MessageTypes.JOIN_GAME:
       case MessageTypes.SPECTATE_GAME:
+      case MessageTypes.PLAYER_TIMEOUT: {
```

Comment:
The only difference between this case and the next case `MAKE_MOVE` seems to be the type logic:

```typescript
type: [GameStatus.GAME_WON_BY_TIMEOUT].includes(game.status)
            ? MessageTypes.GAME_COMPLETE
            : message.type,
```

Can we remove this code and just include `GameStatus.GAME_WON_BY_TIMEOUT` in the logic on line 580

Reply:
We might even be able to combine the logic of `NOTIFY_TIME` case into the `MAKE_MOVE` one below